tests/graph/test_spatial_neighbors.py

```diff
@@ -35,6 +47,21 @@ def test_spatial_neighbors_visium(self, visium_adata: AnnData, n_rings: int, n_n
         if n_rings > 1:
             assert visium_adata.obsp[Key.obsp.spatial_dist()][0].sum() == sum_dist
 
+        # test for library_key
+        visium_adata2 = visium_adata.copy()
+        adata_concat, batch1, batch2 = self._adata_concat(visium_adata, visium_adata2)
+        spatial_neighbors(visium_adata2, n_rings=n_rings)
+        spatial_neighbors(adata_concat, library_key="library_id", n_rings=n_rings)
+        assert adata_concat.obsp[Key.obsp.spatial_conn()][0].sum() == n_neigh
+        np.testing.assert_array_equal(
+            adata_concat[adata_concat.obs.library_id == batch1].obsp[Key.obsp.spatial_conn()].A,
```

Comment:
Prefer `adata_concat.obs['library_id']` (also in other 5 places)

Reply:
done